src/Aspire.Hosting.AWS.DynamoDBLocal/DynamoDBLocalResourceBuilderExtensions.cs

```diff
@@ -0,0 +1,163 @@
+                  // DynamoDB local to use shared db instead of in memory.
+                  .WithVolumeMount($"./dynamodb/{container.Name}", "/home/dynamodblocal/data");
+
+        // Repurpose the WithEnvironment to invoke the users callback to seed DynamoDB local.
+        // This needs a better mechanism to have a callback once a container has been started
+        // so a hosting component can do initial configuration or seeding in the container.
+        containerBuilder.WithEnvironment(context =>
+        {
+            if (context.PublisherName == "manifest")
+            {
+                return;
+            }
+
+            if (container.DisableDynamoDBLocalTelemetry)
+            {
+                // Info on DynamoDB Local telemetry
+                // https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/DynamoDBLocalTelemetry.html
+                context.EnvironmentVariables.Add("DDB_LOCAL_TELEMETRY", "0");
+            }
+        });
+
+        return containerBuilder;
+    }
+
+    /// <summary>
+    /// Add a callback to seed DynamoDB local with tables and data once the container is started.
+    /// </summary>
+    /// <param name="builder"></param>
+    /// <param name="seedDynamoDBCallback"></param>
+    /// <returns></returns>
+    public static IResourceBuilder<IDynamoDBLocalResource> WithSeedDynamoDBCallback(this IResourceBuilder<IDynamoDBLocalResource> builder, Func<SeedDynamoDBUtilities, CancellationToken, Task> seedDynamoDBCallback)
+    {
+        builder.WithEnvironment(context =>
+        {
+            if (context.PublisherName == "manifest")
+            {
+                return;
+            }
+
+            if (!builder.Resource.TryGetAllocatedEndPoints(out var allocatedEndPoints))
+            {
+                throw new DistributedApplicationException("Expected allocated endpoints!");
+            }
+
+            var endpoint = allocatedEndPoints.Single();
+            var serviceUrl = $"http://{endpoint.Address}:{endpoint.Port}/";
+
+            var config = new AmazonDynamoDBConfig
+            {
+                ServiceURL = serviceUrl
+            };
+
+            // TODO: This logic can be simplified to always use the example credentials once I'm able to adjust the
+            // COMMAND used to startup the container. Then I can set the DynamoDB local database to not include the
+            // Access Key ID as part of the file name.
+            AWSCredentials credentials;
+            try
+            {
+                credentials = FallbackCredentialsFactory.GetCredentials();
+            }
+            catch
+            {
+                // DynamoDB local just needs something that looks like credentials. These are not real credentials as you call with the "EXAMPLE" suffix.
+                credentials = new BasicAWSCredentials("AKIAIOSFODNN7EXAMPLE", "wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY");
```

Comment:
I'm worried that if we merge this in its going to get flagged somehow. Is it possible generate a password dynamically here instead of a hardcoded one?

Reply:
This are the stock dummy credentials that we use in the AWS docs. This search on GitHub shows the dummy credentials over 17K times. https://github.com/search?q=AKIAIOSFODNN7EXAMPLE&type=code

My goal is to make sure readers realize this is not an authenticated request with DynamoDB local.

Reply:
LOL. Should be safe then I guess.